.gitlab-ci.yml

```diff
@@ -1,8 +1,76 @@
 stages:
+  - build
+  - install
   - postinstall
   - tests
   - lint
 
+########################################
+# BUILD DEB
+########################################
+
+.build-stage:
+  image: before-install
+  stage: build
+  variables:
+    YNH_BUILD_DIR: "/ynh-build/"
+    YNH_SOURCE: "https://github.com/yunohost"
+  before_script:
+    - DEBIAN_FRONTEND=noninteractive apt --assume-yes -o Dpkg::Options::="--force-confold" install git git-buildpackage postfix python-setuptools
+    - mkdir -p $YNH_BUILD_DIR
+    - cd $YNH_BUILD_DIR
+    - git clone $YNH_SOURCE/$DEB_TO_BUILD
+    - DEBIAN_FRONTEND=noninteractive apt --assume-yes -o Dpkg::Options::="--force-confold" build-dep $YNH_BUILD_DIR/$DEB_TO_BUILD
+  cache:
+    paths: 
+      - /ynh-build/*.deb
+    key: "$CI_JOB_STAGE-$CI_COMMIT_REF_SLUG"
+
+build-yunohost:
+  extends: .build-stage
+  variables:
+    DEB_TO_BUILD: "yunohost"
+  script:
+    - cd $DEB_TO_BUILD
+    - debuild -us -uc
+
+build-ssowat:
+  extends: .build-stage
+  variables:
+    DEB_TO_BUILD: "ssowat"
+  script:
+    - cd $DEB_TO_BUILD
+    - debuild -us -uc
+
+build-moulinette:
+  extends: .build-stage
+  variables:
+    DEB_TO_BUILD: "moulinette"
+  script:
+    - cd $DEB_TO_BUILD
+    - debuild -us -uc
+
+build-metronome:
+  extends: .build-stage
+  variables:
+    DEB_TO_BUILD: "metronome"
+  script:
+    - cd $DEB_TO_BUILD
+    - dpkg-buildpackage -rfakeroot -uc -b -d
```

Comment:
Hmmm I don't know if that's really worth it to build metronome considering the time it takes ? (It's huuuuuuge compared to how often the CI runs) Also if we go this way we should also add python-miniupnpc builds for Buster CI which also is huge. Just having them preinstalled like other dependencies is good enough imho

Reply:
Metronome builds as fast as the other packages on my computer. We'll see how long it takes on the CI.

I don't know, maybe it's completely overkill to build deb on each build, to build deb at every build, but at least it removes the problems when adding/removing dependencies on YNH.

Reply:
On the other hand, I can preinstall all the necessary dependencies for the CI :
https://github.com/YunoHost/yunohost/pull/991/files#diff-96edf7a6f008de9e928d04e1ae5e12a5R19
https://github.com/YunoHost/yunohost/pull/991/files#diff-96edf7a6f008de9e928d04e1ae5e12a5R113

Reply:
I removed the construction of the metronome deb, we can easily add it again, as well as python-miniupnpc